From f4d453da4e54fc50c0e0d5ef8034b10e720f0e77 Mon Sep 17 00:00:00 2001
From: Mark Hatle <mark.hatle@amd.com>
Date: Thu, 4 Jun 2026 19:12:57 -0600
Subject: [PATCH] openavnu: ring_rawsock.c: Add missing poll.h

Use of the ppoll function resulted in:
| ../OpenAvnu/lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c:513:27: error: implicit declaration of function 'ppoll'; did you mean 'poll'? [-Wimplicit-function-declaration]
|   513 |                 int ret = ppoll(&pfd, 1, pts, NULL);
|       |                           ^~~~~
|       |                           poll

Add poll.h which defines ppoll.

Upstream-Status: Pending

Signed-off-by: Mark Hatle <mark.hatle@amd.com>
---
 lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/OpenAvnu/lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c b/OpenAvnu/lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c
index f66878c1..828f02fc 100644
--- a/OpenAvnu/lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c
+++ b/OpenAvnu/lib/avtp_pipeline/platform/Linux/rawsock/ring_rawsock.c
@@ -29,6 +29,9 @@ Complete license and copyright information can be found at
 https://github.com/benhoyt/inih/commit/74d2ca064fb293bc60a77b0bd068075b293cf175.
 *************************************************************************************************************/
 
+#define _GNU_SOURCE
+#include <poll.h>
+
 #include "ring_rawsock.h"
 #include "simple_rawsock.h"
 #include <linux/if_packet.h>
-- 
2.34.1

